From 6c1ea4a051c3dad94de37fe766388ac0a81eb22c Mon Sep 17 00:00:00 2001
From: Mike Westerhof <mwester@dls.net>
Date: Fri, 25 Jul 2008 23:06:06 +0100
Subject: [PATCH] gta01-gps-power-state-resume-preserve.patch

Created an attachment (id=593) / from BZ#79
Updated patch to preserve the power state of the GPS on the GTA01

Currently the GPS is powered up after a suspend/resume occurs.  This patch will
only power the GPS back up if it was powered up when the GTA01 suspended.

Signed-off-by: Mike Westerhof <mwester@dls.net>
---
 arch/arm/plat-s3c24xx/neo1973_pm_gps.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-s3c24xx/neo1973_pm_gps.c b/arch/arm/plat-s3c24xx/neo1973_pm_gps.c
index d020f8d..9c6adfa 100644
--- a/arch/arm/plat-s3c24xx/neo1973_pm_gps.c
+++ b/arch/arm/plat-s3c24xx/neo1973_pm_gps.c
@@ -514,8 +514,8 @@ static int gta01_pm_gps_resume(struct platform_device *pdev)
 {
 #ifdef CONFIG_MACH_NEO1973_GTA01
 	if (machine_is_neo1973_gta01()) {
-		/* FIXME */
-		gps_power_sequence_up();
+		if (neo1973_gps.power_was_on)
+			gps_power_sequence_up();
 	}
 #endif /* CONFIG_MACH_NEO1973_GTA01 */
 
-- 
1.5.6.3

